src/pyfmi/fmi3.pyx

```diff
@@ -2173,6 +2209,198 @@ cdef class FMUModelBase3(FMI_BASE.ModelBase):
 
         return self._derivatives_states_dependencies_kind, self._derivatives_inputs_dependencies_kind
 
+    def _get_directional_proxy(self, var_ref, func_ref, group = None, add_diag = False, output_matrix = None):
+        cdef list data = [], row = [], col = []
+        cdef list local_group
+        cdef int nbr_var_ref  = len(var_ref), nbr_func_ref = len(func_ref)
+        cdef np.ndarray[FMIL3.fmi3_float64_t, ndim=1, mode='c'] v = np.zeros(nbr_var_ref, dtype = np.double)
+        cdef np.ndarray[FMIL3.fmi3_float64_t, ndim=1, mode='c'] data_local
+        cdef int ind_local = 5 if add_diag else 4
```

Comment:
can we add a comment on this? just seems weird first time you encounter it

Reply:
See latest commit; does that help a little?

Not sure how much to add here; if you look for the usage within this function, the role is actually quite clear.